### PR TITLE
#3689-Added Functionality to Sort Based On Transcribed Pages

### PR DESCRIPTION
**Pull request by symmetrically**

### CODE REVIEW

**Comment by saracarl** [comment]
I'm not sure what went wrong here, but there's 2 issues:

1)  The works listed at the top of the collection are 100% transcribed, 0% reviewed.  That's the current functionality, not the desired functionality outlined in the issue which is untranscribed/less transcribed works first.
image

2)  It took forever for my test project to load:
image
which implies that something is being calculated.  I think you should be able to get this with an approach along the lines of join_to work_statistics and order_by
(and it definitely shouldn't take this long.)

**Review by saracarl — CHANGES_REQUESTED** [review]
see comments.

**Comment by saracarl** [comment]
@symmetrically this is still in the incorrect order.  
image
It's much faster, though.

Here's the collection I'm testing with (on my DB it has a nice spread of completedness):  http://localhost:3000/adah/civil-war-and-reconstruction-governors-of-alabama?page=1
Can you include a screenshot of the results before I review this again?

**Comment by benwbrum** [comment]
The most recent commit is an improvement, but it does not distinguish between works that are 0% complete _because they have pages that need review_ and works that are 0% complete because they have not been started yet.

See this example of the top of a collection, looking at the percentage statements in the screenshot.
Screenshot from 2023-08-18 11-03-20

I suspect that this could be fixed by sorting on two columns instead of one: percentage complete `DESC` (as currently), followed by a count of pages needing review `ASC`.

**Comment by benwbrum** [comment]
I've recorded a video, which uses a new collection and these sample documents from the Internet Archive:
* https://archive.org/details/cafrmph_000010
* https://archive.org/details/cafrmph_000013
* https://archive.org/details/cafrmph_000112

https://github.com/benwbrum/fromthepage/assets/199961/486cbd0a-ded5-42ae-ae02-b13013f24907

**Comment by saracarl** [comment]
This will also close #3367

**Review by benwbrum — CHANGES_REQUESTED** [review]

File: db/migrate/20230821140126_add_transcribed_percentage_to_work_statistics.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class AddTranscribedPercentageToWorkStatistics < ActiveRecord::Migration[6.0]
+  def up
+    add_column :work_statistics, :transcribed_percentage, :integer
```
[the remark this answers is not in the document]
Comment (benwbrum) [reply db/migrate/20230821140126_add_transcribed_percentage_to_work_statistics.rb:3]: Once we have the two percentage points, we should sort the list of works based on percent completed (ascending), percent transcribed (ascending) and needs_review_percentage (descending)

**Pull request merged**